### PR TITLE
Fix(eos_designs): Wrong passive interfaces rendered under OSPF process

### LINKED ISSUE #1892: router_ospf.j2 generates incorrect interface configuration
### Issue Summary

Using the ospf and l3_interfaces for multiple switches in the same site to generate OSPF configuration results in the wrong interfaces being assigned to router_ospf on the switches.

This is caused by the incorrect use of arista.avd.natural_sort in [router-ospf.j2#L21](https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2#L21)

As per the docs, the three lists interfaces, ip_addresses and nodes must be populated in consistent order. The above only sorts nodes, which results in incorrect indexing of the interfaces list when generating the router_ospf YaML. Removing the sort fixes the issue

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
Input YaML:

        ospf:
          enabled: true
          nodes:
          - LEAF3A
          - LEAF3B
        l3_interfaces:
        - interfaces:
          - Ethernet4
          - Ethernet6
          - Ethernet5
          ip_addresses:
          - 9.9.9.1/30
          - 9.9.9.5/30
          - 9.9.9.13/30
          nodes:
          - LEAF3A
          - LEAF3B
          - LEAF3A
          ospf:
            enabled: true
            point_to_point: true
            area: 0

Resulting structured_config for LEAF3A:
router_ospf:
  process_ids:
    1:
      vrf: Tenant_A_OP_Zone
      ...
      no_passive_interfaces:
      - Ethernet4
      - Ethernet6
      redistribute:
      ...

Resulting structured_config for LEAF3B:
router_ospf:
  process_ids:
    1:
      vrf: Tenant_A_OP_Zone
      ...
      no_passive_interfaces:
      - Ethernet5
      redistribute:
      ...
```

### Relevant log output

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

Removes an arista.avd.natural_sort for handling ospf passive interfaces which was causing incorrect config to be rendered. Added molecule example to test for incorrect behavior and verified fix.

## Related Issue(s)

Fixes #1892

## Component(s) name

`arista.avd.eos_designs`

## How to test

Tested with molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM, thanks for fixing this quickly!

**Review by tgodaA — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 1]
```diff
@@ -18,7 +18,7 @@ router_ospf:
 {%                 set ospf_router_id = vrf.ospf.router_id | arista.avd.default(switch.router_id) %}
 {%                 for l3_interface in vrf.l3_interfaces | arista.avd.natural_sort %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2:19]: I don't think we need this sorting as well. It's not wrong but I don't see any essence. Should we fix it (if agreed) here or in some other PR.

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2:19]: If it's not breaking anything, let's not change it!

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2:19]: At least not in this PR!

**Pull request merged**